app/src/main/java/org/eyeseetea/malariacare/data/database/datasources/CredentialsDataSource.java

```diff
@@ -0,0 +1,40 @@
+package org.eyeseetea.malariacare.data.database.datasources;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.preference.PreferenceManager;
+
+import org.eyeseetea.malariacare.R;
+import org.eyeseetea.malariacare.domain.boundary.repositories.ICredentialsRepository;
+import org.eyeseetea.malariacare.domain.entity.Credentials;
+
+public class CredentialsDataSource implements ICredentialsRepository {
```

Comment:
@manuelplazaspalacio we had talked on some pair programming with @idelcano avoid create repository interface and the implementation as a data source. We had talked from now create always repository interface and the implementation as a repository and create LocalDataSource , can you refactor it?